apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/StorageClientImpl.ModifyBucketLabels.cs

```diff
@@ -24,38 +27,151 @@ public sealed partial class StorageClientImpl
         /// <inheritdoc />
         public override IDictionary<string, string> ModifyBucketLabels(string bucket, IDictionary<string, string> labels, ModifyBucketLabelsOptions options = null)
         {
-            // TODO: Implement:
-            // - Fetch bucket with GetBucket
-            // - Modify the in-memory dictionary (and remember old values)
-            // - Call PatchBucket pass in the metageneration
-            // - Repeat from scratch up to N times as per options if we get conflicts.
-
-            // In theory, many label modifications could be performed with just a single call rather than
-            // read-modify-write, but:
-            // - Always doing a read-modify-write simplifies the code
-            // - This way we get to return the previous values as well
-            throw new NotImplementedException();
+            ValidateBucketName(bucket);
+            GaxPreconditions.CheckNotNull(labels, nameof(labels));
+
+            int retries = options?.Retries ?? ModifyBucketLabelsOptions.DefaultRetries;
```

Comment:
I wondered about that. Elsewhere we validate things in a `ModifyRequest`, but we don't have anything like that. I'll check that it's non-negative within the property, but put no upper bound on it... a huge upper bound may not be *sensible*, but it at least has a valid meaning...